src/modules/user/user.controller.ts

```diff
@@ -428,6 +430,98 @@ export class UserController {
     }
   }
 
+  @UseGuards(ThrottlerGuard)
+  @Post('/unblock-account')
+  @HttpCode(200)
+  @ApiOperation({
+    summary: 'Request account unblock',
+  })
+  @Public()
+  async requestAccountUnblock(
+    @Body() body: RequestAccountUnblock,
+    @Res({ passthrough: true }) res: Response,
+  ) {
+    try {
+      return await this.userUseCases.sendAccountUnblockEmail(body.email);
+    } catch (err) {
+      if (err instanceof NotFoundException) {
+        throw err;
+      }
+
+      new Logger().error(
+        `[USERS/UNBLOCK_ACCOUNT_REQUEST] ERROR: ${
+          (err as Error).message
+        }, BODY ${JSON.stringify({
+          ...body,
+          user: { email: body.email },
+        })}, STACK: ${(err as Error).stack}`,
+      );
+
+      throw new InternalServerErrorException();
+    }
+  }
+
+  @UseGuards(ThrottlerGuard)
+  @Put('/unblock-account')
+  @HttpCode(200)
+  @ApiOperation({
+    summary: 'Resets user error login counter to unblock account',
+  })
+  @Public()
+  async accountUnblock(@Query('token') token: string) {
+    let decodedContent: {
+      payload: { uuid: string; action: string; email: string };
+    };
+
+    try {
+      const decoded = verifyToken(token, getEnv().secrets.jwt);
+
+      if (typeof decoded === 'string') {
+        throw new ForbiddenException();
+      }
+
+      decodedContent = decoded as {
+        payload: { uuid: string; action: string; email: string };
+      };
+    } catch (err) {
+      throw new ForbiddenException();
+    }
+
+    if (
+      !decodedContent.payload ||
+      !decodedContent.payload.action ||
+      !decodedContent.payload.uuid ||
+      !decodedContent.payload.email ||
+      decodedContent.payload.action !== AccountTokenAction.Unblock ||
+      !validate(decodedContent.payload.uuid)
+    ) {
```

Comment:
Followed same pattern as the recovery flow, however, we can move the token verification to the useCase if needed.